controllers/appsv1/deployment_webhook.go

```diff
@@ -53,7 +53,7 @@ type deploymentInterceptor struct {
 // Handle adds a label to a generated pod if deployment or namespace provide annotaion
 func (d *deploymentInterceptor) Handle(ctx context.Context, req admission.Request) admission.Response {
 	tracer := otel.GetTracerProvider().Tracer(v1.ReconciliationTracer)
-	ctx, span := tracer.Start(ctx, "reconcileDeployment")
+	_, span := tracer.Start(ctx, "reconcileDeployment")
```

Comment:
Does `tracer.Start` modify or extend the given context in some way? Iam asking because we might lose information by using the old ctx, right?

Reply:
At least some implementations modify the context by `calling context.WithValue(...)`.

I believe the new code is better than the old code.  The old code ignored the value reported returned into `ctx`, but it wasn't obvious upon reading it.  The new code makes it clear the modified context gets thrown away.

If we need to re-use the context, we should do that in a follow-up, not in this PR which is only to introduce `golangci-lint`.

Reply:
> The new code makes it clear the modified context gets thrown away.

This is a large method and somebody editing the code might not notice that the current context is thrown away. In this case a nolint annotation would be better.

Reply:
The ctx down in this function is propagated to other calls which might generate spans.